dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/ChatCompletions/Converters/ChatClientAgentRunOptionsConverter.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Agents.AI.Hosting.OpenAI.ChatCompletions.Models;
+using Microsoft.Extensions.AI;
+
+namespace Microsoft.Agents.AI.Hosting.OpenAI.ChatCompletions.Converters;
+
+internal static class ChatClientAgentRunOptionsConverter
+{
+    private static readonly JsonElement s_emptyJson = JsonDocument.Parse("{}").RootElement;
+
+    public static ChatClientAgentRunOptions BuildOptions(this CreateChatCompletion request)
+    {
+        ChatOptions chatOptions = new()
+        {
+            Temperature = request.Temperature,
+            MaxOutputTokens = request.MaxCompletionTokens,
+            FrequencyPenalty = request.FrequencyPenalty,
+            PresencePenalty = request.PresencePenalty,
+            Seed = request.Seed,
+            TopP = request.TopP,
+            StopSequences = request.Stop?.SequenceList ?? [],
+            ResponseFormat = request.ResponseFormat?.ToChatResponseFormat()
+        };
+
+        if (request.ToolChoice is not null)
+        {
+            chatOptions.ToolMode = request.ToolChoice.ToChatToolMode();
+        }
+
+        if (request.Tools?.Count > 0)
+        {
+            chatOptions.Tools = request.Tools.Select(x => x.ToAITool()).ToList();
+        }
+
+        return new()
+        {
+            ChatOptions = chatOptions
+        };
+    }
+
+    private static ChatResponseFormat ToChatResponseFormat(this ResponseFormat responseFormat)
+    {
+        if (responseFormat.IsText)
+        {
+            return ChatResponseFormat.Text;
+        }
+        if (responseFormat.IsJsonObject)
+        {
+            return ChatResponseFormat.Json;
+        }
+        if (responseFormat.IsJsonSchema)
+        {
+            var schema = responseFormat.JsonSchema.JsonSchema;
+            return ChatResponseFormat.ForJsonSchema(schema.Schema, schema.Name, schema.Description);
+        }
+
+        throw new ArgumentOutOfRangeException(nameof(responseFormat));
```

Comment:
Similar question for all of these, whether we want to throw or just default to something reasonable, e.g. like Text here